### PR TITLE
sink (ticdc): support base64 encoded password for mysql sink.

### LINKED ISSUE #7693: Support base64 encoded password for mysql sink.
### Is your feature request related to a problem?

as title.

### Describe the feature you'd like

as title.

### Describe alternatives you've considered

_No response_

### Teachability, Documentation, Adoption, Migration Strategy

_No response_

### PR DESCRIPTION
**Pull request by asddongmen**
### What problem does this PR solve?

Issue Number: close #7693 

### What is changed and how it works?

After this pr, user can use base64 ecnoded password in `sink-uri`.
For example, we've got a dowstream tidb user name  `normal` and password `123456`, we can specify `sink-uri` like:
```shell
--sink-uri="mysql://user:123456=@localhost:4000"
```
or we can use firstly encode the password by base64:
```shell
➜  ~ echo -n '123456' | base64
MTIzNDU2

--sink-uri="mysql://user:MTIzNDU2=@localhost:4000"
```

In the example above, the base64 version is:
```
➜  ~ base64 --version
base64 (GNU coreutils) 8.22
Copyright (C) 2013 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Written by Simon Josefsson.
```
#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)

### Release note 

```release-note
Support base64 encoded password for mysql sink.
```

### CODE REVIEW

File: pkg/sink/mysql/db_helper.go [hunk 1]
```diff
@@ -58,6 +58,8 @@ func GenerateDSN(ctx context.Context, sinkURI *url.URL, cfg *Config, dbConnFacto
 		username = "root"
 	}
 	password, _ := sinkURI.User.Password()
+	password = dmutils.DecryptOrPlaintext(password)
```
Comment (lance6716) [inline pkg/sink/mysql/db_helper.go:61]: maybe can move it to top level `pkg` rather than stay in `dm/pkg`

File: pkg/sink/mysql/db_helper.go [hunk 2]
```diff
@@ -235,3 +238,27 @@ func checkTiDBVariable(ctx context.Context, db *sql.DB, variableName, defaultVal
...
+	password := dbConfig.Passwd
+	if dbConnFactory == nil {
+		dbConnFactory = CreateMySQLDBConn
+	}
+	testDB, err := dbConnFactory(ctx, dbConfig.FormatDSN())
+	if err != nil {
+		// If access is denied and password is encoded by base64, try to decoded password.
+		if mysqlErr, ok := errors.Cause(err).(*dmysql.MySQLError); ok && mysqlErr.Number == tmysql.ErrAccessDenied {
+			if dePassword, decodeErr := base64.StdEncoding.DecodeString(password); decodeErr == nil && string(dePassword) != password {
+				log.Info("fizz decode password", zap.String("password", password), zap.String("decode password", string(dePassword)))
```
Comment (nongfushanquan) [inline pkg/sink/mysql/db_helper.go:252]: Please don't record the password in the log file

Comment (asddongmen) [reply pkg/sink/mysql/db_helper.go]: done

**Review by nongfushanquan — APPROVED** [review]

**Pull request merged**